### DISCUSSION TITLE
How to write a DXF document into a text- or binary string?

### DISCUSSION THREAD

**Discussion by qilong777** [mozman/ezdxf#485] [Q&A] [answered]
I just want Python to generate the corresponding DXF content from the data and return it to Node. If not, maybe I'll make my own

**Comment by qilong777** [comment]
Generating files can be a performance drain because of the amount of data.

**Comment by mbway** [comment]
@qilong777 do you mean like being able to write to a `BytesIO` or `StringIO` to avoid writing to the filesystem? It looks like that is already supported with:

This code creates an ASCII DXF file:
```python
out = io.StringIO()
doc.write(out)
content: str = out.getvalue()

# convert string "content" into bytes if required:
# this applies the required output encoding: doc.output_encoding
# and the correct DXF Unicode encoding style: "\U+xxxx"
data: bytes = doc.encode(content)
```

This code creates a Binary DXF file without going through `StringIO`.  Binary DXF files are not as well supported by 3rd party libraries as ASCII DXF files!
```python
out = io.BytesIO()
doc.write(out, fmt='bin')
out.seek(0)
data: bytes = out.read()
```

**Reply by qilong777** [reply]
Thank you, sorry for taking so long to reply.

**Reply by qilong777** [reply]
yeah,I just want binary data

**Comment by mozman** [comment, marked as answer]
There is a method to create base64 encoded data from a DXF document:
https://ezdxf.mozman.at/docs/drawing/drawing.html#ezdxf.document.Drawing.encode_base64

```Python

# return data to browser ...
return b'data:application/octet-stream;base64,' + doc.encode_base64()

```

**Reply by qilong777** [reply]
Thank you, sorry for taking so long to reply.

**Reply by qilong777** [reply]
Thank you. Your library is very useful
